lib/rspec/core/shared_example_group.rb

```diff
@@ -213,16 +213,35 @@ def valid_name?(candidate)
 
         def warn_if_key_taken(context, key, new_block)
           existing_module = shared_example_groups[context][key]
-
           return unless existing_module
 
-          RSpec.warn_with <<-WARNING.gsub(/^ +\|/, ''), :call_site => nil
-            |WARNING: Shared example group '#{key}' has been previously defined at:
-            |  #{formatted_location existing_module.definition}
-            |...and you are now defining it at:
-            |  #{formatted_location new_block}
-            |The new definition will overwrite the original one.
-          WARNING
+          old_definition = formatted_location existing_module.definition
+          new_definition = formatted_location new_block
+          loaded_spec_files = RSpec.configuration.loaded_spec_files
+
+          if loaded_spec_files.include?(new_definition) && old_definition == new_definition
+            RSpec.warn_with <<-WARNING.gsub(/^ +\|/, ''), :call_site => nil
+              |WARNING: Shared example group '#{key}' has been previously defined at:
+              |  #{old_definition}
+              |...and you are now defining it at:
+              |  #{new_definition}
+              |The new definition will overwrite the original one.
+              |
+              |Your shared example group was automatically loaded because of the
+              |naming of the file, and required specifically somewhere else. To
+              |fix this warning, either remove `_spec` from the file name in which
```

Comment:
The pattern is configurable, and `_spec.rb` might not match what they have configured the pattern to.  Instead, I'd say something like:

> Your shared example group was automatically loaded by RSpec because the file name
> matches the configured pattern (#{config.pattern}), and is also being required from
> somewhere else.  To fix this warning, either rename the file to not match the pattern, or
> do not explicitly require the file.